src/containers/change-password/store/slice.ts

```diff
@@ -0,0 +1,114 @@
+import { Action, createSlice, PayloadAction } from '@reduxjs/toolkit'
+
+import { Status } from 'store/types'
+
+export type ChangePasswordState = {
+  confirmCredentials: {
+    status?: Status
+  }
+  changePassword: {
+    status?: Status
+  }
+  currentPage: Page
+}
+
+export enum Page {
+  CONFIRM_CREDENTIALS = 0,
+  NEW_PASSWORD = 1,
+  LOADING = 2,
+  SUCCESS = 3,
+  FAILURE = 4
+}
+
+const initialState: ChangePasswordState = {
+  confirmCredentials: {
+    status: undefined
+  },
+  changePassword: {
+    status: undefined
+  },
+  currentPage: Page.CONFIRM_CREDENTIALS
+}
+
+const slice = createSlice({
+  name: 'application/ui/changePassword',
+  initialState,
+  reducers: {
+    confirmCredentials: (
+      state,
+      _action: PayloadAction<{ email: string; password: string }>
+    ) => {
+      return {
+        ...state,
```

Comment:
same comments here for this PR around whether we should be doing the immutablejs style of state transform that we're allowed to. maybe we should take this to product eng sync

Reply:
Oh yeah, old habits die hard whoops